cves/CVE-2015-8218.yml

```
@@ -90,14 +97,12 @@ fixes_vcc_instructions: |
   find interesting in there.
 fixes:
 - commit: d4a731b84a08f0f3839eaaaf82e97d8d9c67da46
-  note: ''
-- commit: ''
-  note: ''
+  note: 'Main fix commit'
+- commit: 'a7bbb7fb884a02a7b380ef7afa787fca756b9d82'
+  note: 'Cherry picked commit for version 2.8.2'
 vccs:
 - commit: b2e95e012c3c857eb1f8fddc7a41c0c4f0581f30
```

```suggestion
- commit: 38025e6898ef7caa076ff20daf4fc9f27269e4df
  note: First introduction of bug
- commit: a4f9bb228bb34f0cfa4b55b207c19604b1e36818
  note: Bug pasted a second time
- commit: b2e95e012c3c857eb1f8fddc7a41c0c4f0581f30
  note: 'identified by archeogit, refactoring (DRY)'
```

This is definitely a refactoring commit, from `decode_group3_1d_line` into `ff_ccitt_unpack_init`. Check out the file history: https://github.com/FFmpeg/FFmpeg/commits/master/libavcodec/faxcompr.c

A good hint is that a big segment was added, but an identical segment was also removed. Make sure if this suggestion is auto-merged to include the one below it, as it will removed the duplicate `note:` tag.

---

This is actually really interesting - this comment led me to check a second time, and it appears you are correct that the original code didn't have the bounds check, however even older versions of the code seem to have. Specifically it seems that when the vulnerable methods were expanded the vulnerability was introduced, and then copy-pasted a second time, and then copy-pasted into the extracted helper method